dadi/lib/model/index.js

```diff
@@ -777,6 +780,41 @@ Model.prototype.validateAccess = function ({
   })
 }
 
+/**
+ * Validates a query object and returns an object with `success`
+ * indicating whether validation has failed or passed, and an
+ * `errors` array with any resulting validation errors.
+ *
+ * @param  {Object} query
+ * @return {Object}
+ */
+Model.prototype.validateQuery = function (query) {
+  let response = {
+    success: true,
+    errors: []
+  }
+
+  if (!Array.isArray(query) && Object(query) !== query) {
+    response.success = false
+    response.errors.push({
+      message: 'Query must be either a JSON array or a JSON object.'
+    })
+
+    return response
+  }
+
+  Object.keys(query).every(key => {
+    if (key === '$where') {
+      response.success = false
+      response.errors.push({
+        message: 'Bad query'
```

Comment:
I realise this is legacy, but perhaps we should say exactly why it's a Bad Query? I never liked this blunt response!

Reply:
👍